dotnet-authserver/src/TeacherIdentity.AuthServer/Journeys/CoreSignInJourney.cs

```diff
@@ -106,11 +106,13 @@ public override string GetLastAccessibleStepUrl(string? requestedStep)
         (Steps.EmailConfirmation, _) => Steps.Email,
         (Steps.ResendEmailConfirmation, _) => Steps.EmailConfirmation,
         (Steps.EmailExists, _) => Steps.EmailConfirmation,
-        (Steps.Phone, _) => Steps.EmailConfirmation,
+        (Steps.Phone, { EmailAddressVerified: true }) => Steps.Email,
+        (Steps.Phone, { EmailAddressVerified: false }) => Steps.EmailConfirmation,
         (Steps.PhoneConfirmation, _) => Steps.Phone,
         (Steps.ResendPhoneConfirmation, _) => Steps.PhoneConfirmation,
         (Steps.PhoneExists, _) => Steps.PhoneConfirmation,
-        (Steps.Name, _) => Steps.PhoneConfirmation,
+        (Steps.Name, { MobileNumberVerified: true }) => Steps.Phone,
+        (Steps.Name, { MobileNumberVerified: false }) => Steps.PhoneConfirmation,
```

Comment:
This ensures that the back link on Name always has a valid page to go to. If MobileNumberVerified == true, CanAccessStep(PhoneConfirmation) is false.